TELCODOCS-959: Openshift on AWS Outpost support (#55427)
openshift/openshift-docs: modules/installation-custom-aws-vpc.adoc

====
* Create a public and private subnet in the AWS Region for each availability zone that your control plane uses. Each availability zone can contain no more than one public and one private subnet in the AWS region. For an example of this type of configuration, see link:https://docs.aws.amazon.com/vpc/latest/userguide/VPC_Scenario2.html[VPC with public and private subnets (NAT)] in the AWS documentation.
+
To create a private subnet in the AWS Outposts, you need to first ensure that the Outpost instance is located in the desired availability zone. Then, you can create the private subnet within that availability zone within the Outpost instance, by adding the Outpost ARN. Make sure there is another public subnet in the AWS Region created in the same availability zone.
+

> Record each subnet ID. Completing the installation requires that you enter all the subnets IDs, created in the AWS Region, in the `platform` section of the `install-config.yaml` file and changing the workes machineSet to use the private subnet ID created in the Outpost. See link:https://docs.aws.amazon.com/managedservices/latest/userguide/find-subnet.html[Finding a subnet ID] in the AWS documentation.

Reviewer: ```suggestion
Record each subnet ID. Completing the installation requires that you enter all the subnets IDs, created in the AWS Region, in the `platform` section of the `install-config.yaml` file and changing the workers `machineSet` to use the private subnet ID created in the Outpost. See link:https://docs.aws.amazon.com/managedservices/latest/userguide/find-subnet.html[Finding a subnet ID] in the AWS documentation.
```

Author: Thanks @skrthomas I've applied your feedback.

+
[IMPORTANT]
====

> In case you need to create public subnet in the AWS Outposts, you need to make sure this subnet is not going to be used for the Network/Classic LoadBalancer, otherwise the LoadBalancer creation would fail. To achieve that, the `kubernetes.io/cluster/.*-outposts: owned` special tag must be included in the subnet.

Reviewer: ```suggestion
In case you need to create public subnet in the AWS Outposts, verify that this subnet is not used for the Network or Classic LoadBalancer, otherwise the LoadBalancer creation fails. To achieve that, the `kubernetes.io/cluster/.*-outposts: owned` special tag must be included in the subnet.
```

Reviewer: A few minimalism suggestions

Author: Thanks @skrthomas I've applied your feedback.
